enterprise_subsidy/apps/fulfillment/tests/test_api.py

```diff
@@ -0,0 +1,202 @@
+
+    @mock.patch("enterprise_subsidy.apps.content_metadata.api.ContentMetadataApi.get_content_summary")
+    def test_create_allocation_payload(self, mock_get_content_summary):
+        """
+        Ensure basic happy path of `_create_allocation_payload`
+        """
+        content_summary = {
+            'content_uuid': 'course-v1:edX-test-course',
+            'content_key': 'course-v1:edX-test-course',
+            'source': 'edX',
+            'mode': 'verified',
+            'content_price': 10000,
+            'geag_variant_id': str(uuid4()),
+        }
+        mock_get_content_summary.return_value = content_summary
+        tx_metadata = {
+            'geag_first_name': 'Donny',
+            'geag_last_name': 'Kerabatsos',
+            'geag_date_of_birth': '1900-01-01',
+            'geag_terms_accepted_at': '2021-05-21T17:32:28Z',
+        }
+        transaction = TransactionFactory.create(
+            state=TransactionStateChoices.PENDING,
+            quantity=-19998,
+            ledger=self.subsidy.ledger,
+            lms_user_id=self.lms_user_id,
+            content_key=self.content_key,
+            metadata=tx_metadata,
+        )
+        # pylint: disable=protected-access
+        geag_payload = self.geag_fulfillment_handler._create_allocation_payload(transaction)
+        assert geag_payload.get('order_items')[0].get('productId') == content_summary.get('geag_variant_id')
+
+    def test_validate_pass(self):
+        """
+        Ensure `_validate` method passes
+        """
+        tx_metadata = {
+            'geag_first_name': 'Donny',
+            'geag_last_name': 'Kerabatsos',
+            'geag_date_of_birth': '1900-01-01',
+            'geag_terms_accepted_at': '2021-05-21T17:32:28Z',
+        }
+        transaction = TransactionFactory.create(
+            state=TransactionStateChoices.PENDING,
+            quantity=-19998,
+            ledger=self.subsidy.ledger,
+            lms_user_id=self.lms_user_id,
+            content_key=self.content_key,
+            metadata=tx_metadata,
+        )
+        # pylint: disable=protected-access
+        assert self.geag_fulfillment_handler._validate(transaction)
+
+    def test_validate_fail(self):
+        """
+        Ensure `_validate` method raises with a missing `geag_terms_accepted_at`
+        """
+        tx_metadata = {
+            'geag_first_name': 'Donny',
+            'geag_last_name': 'Kerabatsos',
+            'geag_date_of_birth': '1900-01-01',
+        }
+        transaction = TransactionFactory.create(
+            state=TransactionStateChoices.PENDING,
+            quantity=-19998,
+            ledger=self.subsidy.ledger,
+            lms_user_id=self.lms_user_id,
+            content_key=self.content_key,
+            metadata=tx_metadata,
+        )
+        with pytest.raises(InvalidFulfillmentMetadataException):
+            # pylint: disable=protected-access
+            self.geag_fulfillment_handler._validate(transaction)
```

Comment:
we should probably check that the omission of any of the required fields raises an exception.  So like, add `geag_terms_accepted_at` to `tx_metadata` and do something like:
```python
for field in the_required_fields:
    transaction.metadata = {key, val for key, val in tx_metadata if key != field}
    with pytest.raises(...):
        self.geag_fulfillment_handler._validate(transaction)
```